### DISCUSSION TITLE
emf to png , can re-rendering text ？

### DISCUSSION THREAD

**Discussion by billpeace** [dlemstra/Magick.NET#1454] [Help]
// Read first frame of gif image
 using var image = new MagickImage(Path.Combine(AppDomain.CurrentDomain.BaseDirectory, "image1.emf"));

 // Save frame as jpg
 string file2 = Path.Combine(AppDomain.CurrentDomain.BaseDirectory, "image1.png");
 image.Settings.BackgroundColor = MagickColors.White;
 image.Write(file2);

![image](https://github.com/dlemstra/Magick.NET/assets/372286/3fa62f0c-038d-4a08-976c-83d64e348606)

**Comment by dlemstra** [comment]
I have no idea what you are trying to ask me.

**Comment by billpeace** [comment]
The text has become darker or bolder, Compare with the display in word

**Comment by dlemstra** [comment]
Have you tried setting the `Density` in the `MagickReadSettings`?

**Comment by billpeace** [comment]
As described in the link
      when I open up the emf image in MS Paint, it looks the same as the converted jpg.

I have uploaded the emf file, can you give me a sample code?
thanks very much

https://imagemagick.org/discourse-server/viewtopic.php?t=21983

![image](https://github.com/dlemstra/Magick.NET/assets/372286/4a5fb47a-1f45-4ecf-a373-5a6af8a0c9fc)


[doc1.zip](https://github.com/dlemstra/Magick.NET/files/12914724/doc1.zip)

**Comment by dlemstra** [comment]
What have you tried so far yourself and where are you stuck?

**Comment by billpeace** [comment]
Test in different ways, Solved by setting a very high density
How to find the right density？

 var settings1 = new MagickReadSettings();
 settings1.Density = new Density(1000, 1000, DensityUnit.PixelsPerInch);
 using var image1 = new MagickImage(fileInfo.FullName, settings1);

 string imagepath1 = fileInfo.FullName.Replace(".emf", ".png");
 image1.Write(imagepath1);

**Comment by dlemstra** [comment]
You will need to guess/test that? Take your images and then check at which density you get a consistent result?
